README.md

```diff
@@ -1 +1,209 @@
-# RDF-Connect protobuf files
+# RDF-Connect Protobuf Files
+
+RDF-Connect is a distributed RDF (Resource Description Framework) processing system that enables streaming data processing with real-time orchestration and logging capabilities.
+
+Full specification detailing the implementations can be found on [here](https://rdf-connect.github.io/specification/).
+
+## Core Capabilities
+
+### 🚀 **Distributed Processing Architecture**
+- **Orchestrator-Runner Model**: Centralized orchestration with distributed execution
+- **Processor Lifecycle Management**: Dynamic processor initialization, configuration, and lifecycle control
+- **Bidirectional Streaming**: Full-duplex communication between orchestrators and runners
+
+### 📡 **Advanced Message Routing**
+- **Channel-Based Routing**: Messages routed through logical channels with sequence numbering
+- **Tick Sequencing**: Built-in message ordering and deduplication support
+- **URI-Based Identification**: Unique component identification system
+
+### 🌊 **Streaming Data Protocol**
+- **Chunked Streaming**: Efficient streaming of large datasets with identification
+- **Stream Identification**: Each stream has unique channel and tick identifiers
+- **Data Chunking**: Optimized data transmission with metadata preservation
+
+### 🔧 **Robust Error Handling**
+- **Built-in Error Reporting**: Comprehensive error handling throughout the protocol
+- **Graceful Degradation**: Optional error fields allow for partial success scenarios
+- **Service-Level Error Propagation**: Errors bubble up through the service interface
+
+### 📊 **Pipeline Management**
+- **Turtle Format Support**: Full pipeline information in Turtle RDF serialization
+- **SHACL Integration**: Built-in support for SHACL (Shapes Constraint Language) validation
+- **Dynamic Pipeline Updates**: Runtime pipeline modification capabilities
+
+### 📝 **Comprehensive Logging**
+- **Dedicated Log Streams**: Separate logging channels for debugging and monitoring
+- **Structured Logging**: Log messages with levels, entities, and aliases
+- **Real-time Log Streaming**: Continuous log streaming for live monitoring
+
+## Protocol Components
+
+### Core Message Types
```

Comment:
This listing will need to be updated to signal the changed protobuf definitions.
Also the diagrams and service interface further below will need an update.